cel/library.go

```diff
@@ -1179,7 +1186,7 @@ func (u *uri) scheme() bool {
 				break
 			}
 		}
-		if u.str[u.index] == ':' {
+		if u.peek(':') {
```

Comment:
This fixes the panic for the input "A://". I suggest we adopt `peek` to other implementations where necessary.